habitat-lab/habitat/core/env.py

```diff
@@ -111,6 +111,22 @@ def __init__(
         else:
             self.number_of_episodes = None
 
+        # Filter out inactive sensors from habitat-sim config.
+        if config.gym.cull_visual_sensors:
```

Comment:
Doing this cull logic here in `Env.__init__`... I wonder if this is too late and not ideal. For example, what about our call to `envs.initialize_batch_renderer` which also reads sim_sensors? Will it see the updated sim_sensors? For reference, when I accidentally re-implemented this recently, I put the culling in `default.py patch_config`, which is much earlier in our loading path. I don't know if that's better.

Reply:
I agree. There is some unwrangling to do here.

This pattern of changing configs on-the-fly is causing us headaches with sensors. @aclegg3  and @jturner65 are currently working on refactoring them.

For example:

`rearrange_sim` changes the UUIDs here: https://github.com/facebookresearch/habitat-lab/blob/f41187690ac6607dede2bd50cf50f2360f40b223/habitat-lab/habitat/tasks/rearrange/rearrange_sim.py#L68

UUIDs get changed again when they are passed to sim here: https://github.com/facebookresearch/habitat-lab/blob/f41187690ac6607dede2bd50cf50f2360f40b223/habitat-lab/habitat/sims/habitat_simulator/habitat_simulator.py#L371

@aclegg3 let's consider this as we refactor agents and sensors.

Reply:
So what's your plan here? Are you going to make a change here before merging this PR?

Reply:
Sensor refactor won't be "done" for a few weeks. I suggest you move forward with a stop-gap if you need this change to be around soon. If this is just living on a branch or not really needed in the short term then you could wait. A similar mechanism is planned to be included in the sensor refactor.

Reply:
@eundersander I would then go and merge this as-is.